qt/widgets/instrumentview/src/InstrumentWidgetPickTab.cpp

```diff
@@ -1237,50 +1189,53 @@ void DetectorPlotController::plotSingle(int detid) {
 *   LENGTH
 *   PHI
 * The units can be set with setTubeXUnits(...) method.
-* @param detid :: A detector id. The miniplot will display data for a component
+* @param detindex :: A detector index. The miniplot will display data for a
+* component
 * containing the detector
 *   with this id.
 */
-void DetectorPlotController::plotTube(int detid) {
+void DetectorPlotController::plotTube(size_t detindex) {
   const auto &actor = m_instrWidget->getInstrumentActor();
-  auto &det = actor.getDetectorByDetID(detid);
-  boost::shared_ptr<const Mantid::Geometry::IComponent> parent =
-      det.getParent();
-  Mantid::Geometry::ICompAssembly_const_sptr ass =
-      boost::dynamic_pointer_cast<const Mantid::Geometry::ICompAssembly>(
-          parent);
-  if (parent && ass) {
+  const auto &componentInfo = actor.componentInfo();
+
+  if (!componentInfo.hasParent(detindex)) {
+    m_plot->clearCurve();
+    return;
+  }
+
+  auto parent = componentInfo.parent(detindex);
+  if (componentInfo.numberOfDetectorsInSubtree(parent) > 1) {
```

Comment:
So this plots only if there are two detectors or more?

Also, is it really relevant for performance to not use `detectorsInSubtree` here? Having a extra method just for this single case would not be justified otherwise.